### DISCUSSION TITLE
生产环境打包的疑问？

### DISCUSSION THREAD

**Discussion by FickXu** [yume-chan/ya-webadb#447] [Q&A] [answered]
项目做了少许调整，比如减少菜单栏菜单等。使用rush build命令打包，得到日志文件
`
Invoking: next build 
warn  - You have enabled experimental feature (esmExternals) in next.config.js.
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.

info  - Linting and checking validity of types...
info  - Creating an optimized production build...
info  - Compiled successfully
info  - Collecting page data...
info  - Generating static pages (0/16)
info  - Generating static pages (4/16)
info  - Generating static pages (8/16)
info  - Generating static pages (12/16)
info  - Generating static pages (16/16)
info  - Finalizing page optimization...

Page                                       Size     First Load JS
┌ ○ /                                      1.26 kB         258 kB
├   /_app                                  0 B             257 kB
├ ○ /404                                   240 B           257 kB
├ ○ /bug-report                            3.54 kB         260 kB
├ ○ /device-info                           3.72 kB         260 kB
├ ○ /file-manager                          52 kB           309 kB
├ ○ /framebuffer                           1.4 kB          258 kB
├ ○ /index_copy                            1.98 kB         259 kB
├ ○ /install                               1.03 kB         258 kB
├ ○ /logcat                                3.26 kB         260 kB
├ ○ /packet-log                            1.59 kB         258 kB
├ ○ /power                                 3.11 kB         260 kB
├ ○ /reverse                               696 B           257 kB
├ ○ /scrcpy (307 ms)                       13.9 kB         271 kB
├ ○ /shell                                 114 kB          370 kB
├   └ css/e29414ad228c7c9b.css             663 B
└ ○ /tcpip                                 4.25 kB         261 kB
+ First Load JS shared by all              257 kB
  ├ chunks/framework-664425b42760ddbe.js   42.1 kB
  ├ chunks/main-f96ec7c3bd427523.js        30.9 kB
  ├ chunks/pages/_app-9da0b6f6fa285ecc.js  182 kB
  ├ chunks/webpack-ac6dc2790397f70b.js     2.05 kB
  └ css/b6853f0190ea9f57.css               293 B

○  (Static)  automatically rendered as static HTML (uses no initial props)

`
看信息没有报错，应该是成功了，但是在目录中没有找到对应的静态文件，请问是我的命令使用错误吗？还是有其它的方式呢

**Comment by yume-chan** [comment, marked as answer]
Run `npx next export` in `apps/demo` folder to produce static HTML files.

**EDIT:** added more details here: https://github.com/yume-chan/ya-webadb/blob/main/CONTRIBUTING.md#deploy-demo

**Reply by FickXu** [reply]
谢谢你的及时回复，已经处理好了。
不知道为什么我下载的项目在CONTRIBUTING.md文件中没有#deploy-demo这一项，但是在github中是可以看到的。

**Comment by Naxexia** [comment]
Hi @yume-chan 
i followed this guid "https://github.com/yume-chan/ya-webadb/blob/main/CONTRIBUTING.md#deploy-demo" export static HTML files and push these files to github.io repository , but found the index page shows some errors. Can you help check it?
<img width="1787" alt="image" src="https://user-images.githubusercontent.com/20432988/207590699-71ad1311-e4de-4195-90a7-4f978d500b4a.png">

**Reply by yume-chan** [reply]
If you are deploying to a subpath, set the `BASE_PATH` environment variable:

https://github.com/yume-chan/ya-webadb/blob/8bcd0ab913024612fe1633107269a3e97fdff0d7/.github/workflows/deploy.yml#L10-L11
